Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeInfoVBEExtensions.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Rubberduck.VBEditor.ComManagement.TypeLibs
+{
+    /// <summary>
+    /// An internal interface exposed by VBA for all components (modules, class modules, etc)
+    /// </summary>
+    /// <remarks>This internal interface is known to be supported since the very earliest version of VBA6</remarks>
+    [ComImport(), Guid("DDD557E1-D96F-11CD-9570-00AA0051E5D4")]
+    [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+    public interface IVBEComponent
+    {
+        void Placeholder1();
+        void Placeholder2();
+        void Placeholder3();
+        void Placeholder4();
+        void Placeholder5();
+        void Placeholder6();
+        void Placeholder7();
+        void Placeholder8();
+        void Placeholder9();
+        void Placeholder10();
+        void Placeholder11();
+        void Placeholder12();
+        void CompileComponent();
+        void Placeholder14();
+        IDispatch GetStdModAccessor();
+        void Placeholder16();
+        void Placeholder17();
+        void Placeholder18();
+        void Placeholder19();
+        void Placeholder20();
+        void Placeholder21();
+        void Placeholder22();
+        void Placeholder23();
+        void Placeholder24();
+        void Placeholder25();
+        void Placeholder26();
+        void Placeholder27();
+        void Placeholder28();
+        void Placeholder29();
+        void Placeholder30();
+        void Placeholder31();
+        void Placeholder32();
+        void Placeholder33();
+        void GetSomeRelatedTypeInfoPtrs(out IntPtr a, out IntPtr b);        // returns 2 TypeInfos, seemingly related to this ITypeInfo, but slightly different.
+    }
+
+    /// <summary>
+    /// Exposes the VBE specific extensions provided by an ITypeInfo
+    /// </summary>
+    public class TypeInfoVBEExtensions : IDisposable
+    {
+        private readonly TypeInfoWrapper _parent;
```

Comment:
This is one of four causes for the memory corruption and the one that leads to Excel ghosting, and corrupting VS' debugger. With this as-is, we get:

```
Exception thrown at 0x747C01E8 (clr.dll) in EXCEL.EXE: 0xC0000005: Access violation reading location 0x02D111FC.
The Common Language Runtime cannot stop at this exception. Common causes include: incorrect COM interop marshalling and memory corruption. To investigate further use native-only debugging.
Exception thrown at 0x747C01E8 (clr.dll) in EXCEL.EXE: 0xC0000005: Access violation reading location 0x02D111E4.
The Common Language Runtime cannot stop at this exception. Common causes include: incorrect COM interop marshalling and memory corruption. To investigate further use native-only debugging.
Exception thrown at 0x7764BA99 (ole32.dll) in EXCEL.EXE: 0xC0000005: Access violation reading location 0x02D111E4.
The Common Language Runtime cannot stop at this exception. Common causes include: incorrect COM interop marshalling and memory corruption. To investigate further use native-only debugging.
Exception thrown at 0x747C01E8 (clr.dll) in EXCEL.EXE: 0xC0000005: Access violation reading location 0x02D111E4.
```

Because the `_parent` is only used in 2 exceptions and only its `Name` property is read, it is sufficient to simply cache the `Name` property, rather than the entire `TypeLibWrapper` class. Doing so removes the memory corruption and VS is able to end its debugging session cleanly (though Excel does exist with AVs still from 3 other memory corruptions):

```
Exception thrown at 0x747C01E8 (clr.dll) in EXCEL.EXE: 0xC0000005: Access violation reading location 0x076411FC.
The Common Language Runtime cannot stop at this exception. Common causes include: incorrect COM interop marshalling and memory corruption. To investigate further use native-only debugging.
Exception thrown at 0x747C01E8 (clr.dll) in EXCEL.EXE: 0xC0000005: Access violation reading location 0x076411E4.
The Common Language Runtime cannot stop at this exception. Common causes include: incorrect COM interop marshalling and memory corruption. To investigate further use native-only debugging.
Exception thrown at 0x747C01E8 (clr.dll) in EXCEL.EXE: 0xC0000005: Access violation reading location 0x076411E4.
```